nodejs/awsx/apigateway/apikey.ts

```diff
@@ -0,0 +1,147 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// These APIs are currently experimental and may change.
+
+import * as aws from "@pulumi/aws";
+import * as api from "./api";
+
+export type APIKeySource = "HEADER" | "AUTHORIZER";
+
+/**
+ * Input properties for creating a UsagePlan and associated API Keys.
+ */
+export interface APIKeyArgs {
+
+    /**
+     * Define the apis you would like to associate the usage plan with. This can be used in place of
+     * defining the apiStages defined in the [usagePlan]. You cannot define [apis] and [usagePlan.apiStages].
+     */
+    apis?: api.API[];
+
+    /**
+     * Define the usage plan to create. If an existing Usage Plan is passed in, then the API Keys
+     * will be associated with the usage plan and no [apis] can be added.
+     */
+    usagePlan?: aws.apigateway.UsagePlanArgs | aws.apigateway.UsagePlan;
+
+    /**
+     * The API keys you would like to create & associate with the usage plan.
+     */
+    apiKeys?: Array<aws.apigateway.ApiKeyArgs | aws.apigateway.ApiKey>;
+}
+
+export interface AssociatedAPIKeys {
+    readonly usagePlan: aws.apigateway.UsagePlan;
+    readonly keys: Key[];
+}
+
+export interface Key {
+    apikey: aws.apigateway.ApiKey;
+    usagePlanKey: aws.apigateway.UsagePlanKey;
+}
+
+/**
+ * Helper function that allows you to quickly create API Keys and associate them with an API.
+ *
+ * @param name The _unique_ name of the resource.
+ * @param args The arguments to use to populate this resource's properties.
+ */
+export function createAssociatedAPIKeys(name: string, args: APIKeyArgs): AssociatedAPIKeys {
+    const usagePlan = getUsagePlan(name, args);
+    const keys = getKeys(name, args, usagePlan);
+
+    return {
+        usagePlan: usagePlan,
+        keys: keys,
+    };
+}
+
+/** @internal */
+function getUsagePlan(name: string, args: APIKeyArgs): aws.apigateway.UsagePlan {
+    if (args.usagePlan && args.usagePlan instanceof aws.apigateway.UsagePlan) {
```

Comment:
sounds like you were having trouble with instanceof?  Shoudl this be CustomeResult.IsInstance?